app/assets/javascripts/upload.coffee

```diff
@@ -291,6 +306,16 @@ bulkCorrectionUpload = (fileInput) ->
         console.log(data)
         result = (successful: [data])
         console.log result
+
+        # Disable upload if uploaded (zip)-file is empty
+        # Temporary fix for ":type option required" error in the
+        # app/controllers/submissions_controller.rb -> show_correction
+        if data.metadata.size == 0
+          console.log('empty file encountered during bulk upload')
+          alert($('#bulk-uploadButton-button-actual').data('tr-failure-empty-file'))
+          $('#bulk-uploadButton-button-actual').hide()
+          return
+
```

Comment:
I played around with some sample files and observed the following. First of all, bulk upload means uploading several files at once using a name pattern that MaMpf recognizes (e.g. keep the names from the bulk download). In my tests, I bulk-uploaded corrections for two students. It appears that `data.metadata` contains only information on the file that was uploaded last. So, in my tests, if the zero-size file had a name that came alphabetically before the name of the non-zero-size file, then the UI did not catch the zero-size. (It did if the zero-size file came last). Of course, the server-sided validation caught the zero-size file when the UI did not.

Reply:
Found the bug. The `directUpload` only passed the last uploaded file to the `successCallback`. I fixed this but didn't want to break any other usage of `directUpload`; this is why I've introduced a second parameter for this method which is only used in the callback inside `bulkCorrectionUpload`. This is not really best practice, but if I wanted to fix this more thoroughly, I'd have to go through more parts of the file, which is nothing I like to do in Coffeescript. But we should keep this in mind, convert the `upload.coffee` to JS and greatly clean up, it's really a mess with lots of hardcoded strings, intermingled functionality and lots of states variables throughout the file.

**With the fix in the latest commit, please try to upload files again**. You should also get a message that tells you which file is empty. (If multiple files are empty, you get this info just for one file but can then progressively get rid of the empty files).

Also, the "Cancel" button of the bulk upload does not work, which is something for another PR.